rns_shell/compositor/Compositor.cpp

```diff
@@ -266,17 +266,41 @@ void Compositor::commit(bool immediate=false) {
     }
 
     if(immediate) {
-       RNS_PROFILE_API_OFF("RenderTree Immediate:", renderLayerTree());
-       // Unlock here, after rendering of tree is done for immediate rendering
-       isMutating.unlock();
-       return;
+        RNS_PROFILE_API_OFF("RenderTree Immediate:", renderLayerTree());
+        // Unlock here, after rendering of tree is done for immediate rendering
+        isMutating.unlock();
+        return;
+    }
+
+    //Async RenderTree update
+    //a. If state is idle -> update to schedule state and schedule rendering
+    //b. If state is schedules -> no scheduling required
+    //c. If state is Pending -> not handled in current rendertree solution
```

Comment:
in comment on Point c:, Does pendimg refer inProgress or this would be an additional state in future

Reply:
its InProgress state. Updated.